config.js

```diff
@@ -1,5 +1,9 @@
 export default {
     openAiKey: process.env.OPENAI_API_KEY,
+    azureOpenAIApiKey: process.env.AZURE_OPENAI_API_KEY,
+    azureOpenAIApiInstanceName: process.env.AZURE_OPENAI_API_INSTANCE_NAME,
+    azureOpenAIApiDeploymentName: process.env.AZURE_OPENAI_API_DEPLOYMENT_NAME,
+    azureOpenAIApiVersion: process.env.AZURE_OPENAI_API_VERSION,
```

Comment:
could it not be envs? maybe there are some default values we can use here?
env used for the only purpose not to flash our credentials in the configs and outputs.

also I just noticed azureOpenAI is spelled with all capitals `AI`. and while this is a right way to spell abbreviations, I don't like the way it would look (OpenAIAPI), so please lower the case just like in the `openAiKey`. Also we know this is an API, maybe drop Api from the variables too?

Reply:
because if there is no defaults, we should check they are present in the config too, if you prefer to use azure open ai

Reply:
> could it not be envs? maybe there are some default values we can use here?

There is no default on Azure. LangChain do checking it. When user setting it not enough, LangChain raise exception.

Sample: Case of forget setting AZURE_OPENAI_API_DEPLOYMENT_NAME
```
$ ./autocommit.js 
file:///home/takuya-o/src/git-aicommit/node_modules/langchain/dist/chat_models/openai.js:215
                throw new Error("Azure OpenAI API deployment name not found");
                      ^

Error: Azure OpenAI API deployment name not found
    at new ChatOpenAI (file:///home/takuya-o/src/git-aicommit/node_modules/langchain/dist/chat_models/openai.js:215:23)
    at file:///home/takuya-o/src/git-aicommit/autocommit.js:59:16
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)
```
We don't like this exception, so I will add code that if any settings related to AZURE are set, if there are items that are not set, will display an error.


> also I just noticed azureOpenAI is spelled with all capitals `AI`. and while this is a right way to spell abbreviations, I don't like the way it would look (OpenAIAPI), so please lower the case just like in the `openAiKey`. Also we know this is an API, maybe drop Api from the variables too?

Next, about the naming scheme. It comes from LangChain.

See: [LangChain JS/TS Docs](https://js.langchain.com/docs/getting-started/guide-llm#llms-get-predictions-from-a-language-model)
```
3.2. For Azure OpenAI:
const model = new OpenAI({
  azureOpenAIApiKey: "...",
  azureOpenAIApiInstanceName: "....",
  azureOpenAIApiDeploymentName: "....",
  azureOpenAIApiVersion: "....",
  temperature: 0.9
});
```

However, I will change the naming schema as below it is match with your original code in autocomit.js.
```
const openai = new ChatOpenAI({
    modelName: config.modelName,
    openAIApiKey: config.openAiKey,
    azureOpenAIApiKey: config.azureOpenAiKey,
    azureOpenAIApiInstanceName: config.azureOpenAiInstanceName,
    azureOpenAIApiDeploymentName: config.azureOpenAiDeploymentName,
    azureOpenAIApiVersion: config.azureOpenAiVersion,
    temperature: config.temperature,
    maxTokens: config.maxTokens,
})
```
Of course, config.js does the same.